### ISSUE TITLE
Crash on manual scan for MAME with mameinfo.dat file

### ISSUE THREAD

**Issue by betovidal** [libretro/RetroArch#13026]
## Description

Using retroarch v1.9.9 and MAME (current) core (0.235), manually scanning a folder for roms and specifying the Arcade DAT file mameinfo.dat from [this page](https://mameinfo.mameworld.info/) retroarch crashes and no log information is generated about the crash, even if set to Debug.

### Expected behavior

Retroarch should not crash and MAME playlist should be generated (ideally with the correct names).

### Actual behavior

Retroarch crashes and it doesn't generate a log file that indicates any information about the crash.

### Steps to reproduce the bug

1. Import content -> Manual scan
2. Content directory pointing to ROMS folder
3. System name set to "MAME"
4. Default Core set to "Arcade (MAME - Current)
5. Scan recursively and Scan inside Archives set to "ON"
6. Arcade DAT File pointing to mameinfo.dat file, downloaded from site above
7. Start scan
8. Retroarch crashes

### Version/Commit

* Build Date: Sep 6 2021
* Version 1.9.9
* Git Version: 36888612eb

### Environment information

- OS: Archlinux
- Compiler: GCC(11.1.0) 64-bit (installed through pacman, I didn't compile it myself)

**Comment by LibretroAdmin** [comment]
Is this still an ongoing issue with latest nightlies?

**Comment by c0d3h4x0r** [comment]
Yes, this still happens.  I just encountered it repeatedly last night with the latest Android nightly on MY AYN Odin Plus.  Makes it impossible to create a playlist of MAME games containing real game names rather than zipfile names.
